File: pyodide_projects/oldies/bundle_tutorial/jbook/discover/widget_edit.cpp
#include "hello_imgui/hello_imgui.h"
#include "imgui.h"

struct AppState {
    float globe_size = 100.0f;
};

void Gui(AppState& app_state) {
    HelloImGui::ImageFromAsset("images/world.png", ImVec2(app_state.globe_size, 0.0f));
    bool changed = ImGui::SliderFloat("Globe size", &app_state.globe_size, 20.0f, 200.0f);
}

int main() {
    AppState app_state;
    auto gui_fn = [&]() { Gui(app_state); };
    HelloImGui::Run(gui_fn, "Edit Values", false, false, {220, 220});
}

File: pyodide_projects/oldies/bundle_tutorial/single_page_book_app/jbook/discover/widget_edit.py
from imgui_bundle import imgui, hello_imgui, ImVec2


class AppState:
    globe_size: float = 100.0


def gui(app_state: AppState):
    hello_imgui.image_from_asset("images/world.png", ImVec2(app_state.globe_size, 0))
    _changed, app_state.globe_size = imgui.slider_float("Globe size", app_state.globe_size, 20.0, 200.0)


def main():
    app_state = AppState()
    hello_imgui.run(lambda: gui(app_state), window_title="Edit Values", window_size=(220, 220))


if __name__ == "__main__":
    main()
